### PR TITLE
Reduce compilation time

### PR DESCRIPTION
**Pull request by lumburovskalina**
Issue: 
https://github.com/code-kern-ai/refinery/issues/152 

## Compilation time
Tasks:
- Since version Angular 9, the AOT compiler is used (in the angular.json for the build process we should set this to true)
- Additionally, in the tsconfig.json we should enable the Ivy compiler
- Additional configurations in angular.json (buildOptimization should be set to false and replaced with other options since it is taking too much time) (sourceMap should be also set to true since it is used only for debugging) (vendorChunk, extractLicenses, namedChunks should be set to false)
- Increase Node's memory limit
- browserTarget should be changed to :build:development (changes from Angular 13)
- defaultConfiguration should be set to development

### CODE REVIEW

File: angular.json [hunk 1]
```diff
@@ -60,18 +60,47 @@
...
+              "progress": false,
               "vendorChunk": false,
-              "buildOptimizer": true,
+              "buildOptimizer": false,
+              "subresourceIntegrity": false,
               "budgets": [
                 {
                   "type": "initial",
-                  "maximumWarning": "5mb",
-                  "maximumError": "8mb"
```
Comment (JWittmeyer) [inline angular.json]: not sure we should change these -> only the local build should be affected by the changes, not the drone build so the difference in maximum build size doesn't make sense imo

File: package.json [hunk 2]
```diff
@@ -3,10 +3,10 @@
   "version": "0.0.0",
   "scripts": {
     "ng": "ng",
-    "start": "ng serve",
+    "start": "ng serve --configuration development",
     "start-proxy": "ng serve --proxy-config proxy.conf.json",
     "build": "ng build",
-    "build-prod": "ng build --prod --aot --build-optimizer --outputPath=./dist/out",
```
Comment (JWittmeyer) [inline package.json:9]: see above, for production (drone builds) this shouldn't be changed, right?

Comment (lumburovskalina) [reply package.json:9]: yes, that is correct, I reverted those changes

**Comment by JWittmeyer** [comment]
Optimization looks good, i have improvements over all checked aspects:
start normal ~ 2m 31 sec
start enhanced ~ 3m 20 sec (had some initial install tasks)
start enhanced 2nd ~ 1m 22 sec
change normal html ~ 1m 22 sec
change enhanced html ~ 28 sec
change normal ts ~ 1m 16sec
change enhanced ts ~ 25 sec

**Comment by JWittmeyer** [comment]
Error on projects page:
image

- [x] resolved

**Comment by JWittmeyer** [comment]
Error databrowser:
image

- [x] resolved

**Comment by JWittmeyer** [comment]
Pretty much all pages have either errors or warnings.
Example error Warnings:
image

- [x] resolved


They seem to be related to existing code only noticeable after the change.

**Comment by JWittmeyer** [comment]
Filtering on embedding creation doesnt work
image

- [x] resolved

**Comment by JWittmeyer** [comment]
Crowd labeler:
image

- [x] resolved

**Comment by JWittmeyer** [comment]
Not sure if related but felix mentioned this in the last test as well:
image

Steps to reproduce:
1. Import Clickbait sample
2. create labelingtask on running_id 
3. switch to labeling (data browser looks fine)

- [x] resolved

**Comment by JWittmeyer** [comment]
filtering in embeddings still doesn't work. (no console error but the function doesn't work anymore)

- [x] resolved

**Comment by JWittmeyer** [comment]
crowd heuristic still doesn't work
image

- [x] resolved

**Review by JWittmeyer — APPROVED** [review]

**Pull request merged**